controller/state/interface.go

```diff
@@ -12,7 +12,7 @@ type State interface {
 	Get(ctx context.Context, uuid string) (tasks.Task, error)
 	GetAll(ctx context.Context) ([]tasks.Task, error)
 	GetHead(ctx context.Context) (tasks.RecordUpdate, error)
-	Update(ctx context.Context, uuid string, req tasks.UpdateTask) (tasks.Task, error)
+	Update(ctx context.Context, uuid string, req tasks.UpdateTask, run int) (tasks.Task, error)
```

Comment:
The name "run" doesn't really tell me much if I'm just reading this interface. Perhaps `runCounter`? Another option is to start writing docs for the non-obvious parameters.

Also, did you consider just adding this to UpdateTask?

Reply:
Will rename to `runCount`.  I was not sure if keeping a run count was OK or if a new task was needed, and I for some reason thought this would be easier to change.  It really should be in UpdateTask.